streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -570,6 +581,49 @@ void runLoop() {
         }
     }
 
+    public void setStreamsUncaughtExceptionHandler(final StreamsUncaughtExceptionHandler streamsUncaughtExceptionHandler, final KafkaStreams kafkaStreams) {
```

Comment:
I am looking into using a wrapper and call back so we do not have to introduce a KafkaStreams object here but until I figure that out this will be stop gap.

though this will not work in the rebalance listener as it needs the reference as well. maybe this is the only way idk yet. @cadonna WDYT?